static_frame/core/index.py

```diff
@@ -1226,17 +1226,19 @@ def level_add(self,
 
         if index_constructor is None:
             # cannot assume new depth is the same index subclass
-            index_constructors = (cls_depth, self.__class__)
-        else:
-            index_constructors = (index_constructor, self.__class__) #type: ignore
+            index_constructor = cls_depth
+
+        indices = [index_constructor((level,)), self.copy()]
```

Comment:
Oh good point - this should use `immutable_index_filter(self)`! Since we are returning a new `IndexHierarchy` object, all it's internally managed indices must _not_ be `IndexGO` objects.